vtex-sites/nextjs.store
# Fix: `scrollTo` not working on Safari 15.4

## What's the purpose of this pull request?

This PR attempts to fix two issues:
1.  `scroll-behavior` not working on Safari 15.4 when using `scrollTop`: https://developer.apple.com/forums/thread/703294
![not-scrolling](http://g.recordit.co/FfIeoY9Ail.gif)
2. Not scrolling to the last image position.
![last-image-missing](http://g.recordit.co/6VBDtpqM0v.gif)

## How does it work?

1. This issue could also be fixed g the `scroll-behavior: smooth` property. But after aligning with the design team, we favor the workaround approach with a smooth's effect. We should keep an eye out for safari updates and remove them when the native approach is available.

2. Increased a little the threshold, but it won't work for all the cases. It needs further improvements and should be tackled at another moment.

## How to test it?

1. Open this [preview link](https://sfj-d31738a--nextjs.preview.vtex.app/apple-magic-mouse-99988212/p) on Safari.
2. Click on the Down Arrow button, scroll should work, and you'll see more images' thumbnails. 


## References

- https://developer.apple.com/forums/thread/703294
- https://github.com/WebKit/WebKit/pull/1387
- https://developer.apple.com/documentation/safari-release-notes
- https://developer.mozilla.org/en-US/docs/Web/CSS/::-webkit-scrollbar

## Checklist

<em>You may erase this after checking them all ;)</em>

**Changelog**

**PR Description**

**Documentation**
